### PR TITLE
[fix] Move s3 fs connect outside the lock critical area

### PR DESCRIPTION
**Pull request by platoneko**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

1. Fix potential null client bug in S3FileSystem.
2. Move s3 fs connect outside the lock critical area, which may cause serious lock contention when create tablet.
3. Use S3Conf instead of map to create S3FileSystem.
4. Remove a useless ut.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
5. Has unit tests been added: (Yes/No/No Need)
6. Has document been added or modified: (Yes/No/No Need)
7. Does it need to update dependencies: (Yes/No)
8. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by compiletheworld** [comment]
LGTM

File: be/src/olap/storage_policy_mgr.cpp [hunk 1]
```diff
@@ -24,43 +24,60 @@
...
-        auto s3_fs = std::dynamic_pointer_cast<io::S3FileSystem>(
-                io::FileSystemMap::instance()->get(name));
-        DCHECK(s3_fs);
-        s3_fs->set_ak(policy->s3_ak);
-        s3_fs->set_sk(policy->s3_sk);
-        s3_fs->connect();
-        it->second = std::move(policy);
-    } else {
-        // can't find name's policy, so do nothing.
+void StoragePolicyMgr::update(const std::string& name, const StoragePolicyPtr& policy) {
```
Comment (morningman) [inline be/src/olap/storage_policy_mgr.cpp:27]: Looks like the `update` and `refresh` policy process may be called concurrently?
If so, it is not safe to update `s3_fs` outside the lock

Comment (platoneko) [reply be/src/olap/storage_policy_mgr.cpp:27]: i will fix soon

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**